packages/neos-ui/src/Containers/LeftSideBar/NodeTreeToolBar/Buttons/DeleteSelectedNode/index.js

```diff
@@ -1,42 +1,57 @@
 import React, {PureComponent, PropTypes} from 'react';
 import {connect} from 'react-redux';
+import {$get} from 'plow-js';
+
 import IconButton from '@neos-project/react-ui-components/lib/IconButton/';
+import {selectors, actions} from '@neos-project/neos-ui-redux-store';
 
-@connect()
+@connect(state => ({
+    focusedNodeContextPath: selectors.UI.PageTree.getFocusedNodeContextPathSelector(state),
+    siteNodeContextPath: $get('cr.nodes.siteNode', state),
+    getNodeByContextPath: selectors.CR.Nodes.nodeByContextPath(state)
+}), {
+    removeNode: actions.CR.Nodes.commenceRemoval
+})
 export default class DeleteSelectedNode extends PureComponent {
     static propTypes = {
-        isDisabled: PropTypes.bool,
-        className: PropTypes.string
-    };
+        className: PropTypes.string,
 
-    static defaultProps = {
-        isDisabled: true
+        focusedNodeContextPath: PropTypes.string.isRequired,
+        siteNodeContextPath: PropTypes.string.isRequired,
+        getNodeByContextPath: PropTypes.func.isRequired,
+        removeNode: PropTypes.func.isRequired
     };
 
     constructor(props) {
         super(props);
 
-        this.handleDeleteSelectedNodeClick = this.deleteSelectedNode.bind(this);
+        this.handleRemoveSelectedNodeClick = this.removeSelectedNode.bind(this);
     }
 
     render() {
-        const {
-            isDisabled,
-            className
-        } = this.props;
+        const {className, focusedNodeContextPath, siteNodeContextPath, getNodeByContextPath} = this.props;
+        const node = getNodeByContextPath(focusedNodeContextPath);
+
+        //
+        // Do not allow deletion, when there's no focused node, the focused node is auto created or the focused node
+        // is the site node
+        //
+        const isDisabled = !node || $get('isAutoCreated', node) || siteNodeContextPath === focusedNodeContextPath;
```

Comment:
I agree. Imho this should be moved to `selectors.CR.Nodes`, but I'd postpone that, since this is not such big of a concern for the alpha.